data_import/models.py

```diff
@@ -144,9 +146,30 @@ def raw_table_name(self):
     @property
     def processing(self):
         '''
-        TO-DO: Find a less expensive way to check whether an instance
-        is processing.
+        Inspect the queue for work related to the StandardizedFile
+        at hand. If there is active work, or work on the queue,
+        return True. Otherwise, return False.
         '''
+        from celery.task.control import inspect
+
+        i = inspect()
+
+        active = i.active()
+
+        for worker, import_tasks in active.items():
+            for task in import_tasks:
+                kw_args = json.loads(task['kwargs'])
```

Comment:
I'm getting this traceback:

```
Traceback:

File "/usr/local/lib/python3.5/site-packages/django/core/handlers/exception.py" in inner
  34.             response = get_response(request)

File "/usr/local/lib/python3.5/site-packages/django/core/handlers/base.py" in _get_response
  115.                 response = self.process_exception_by_middleware(e, request)

File "/usr/local/lib/python3.5/site-packages/django/core/handlers/base.py" in _get_response
  113.                 response = wrapped_callback(request, *callback_args, **callback_kwargs)

File "/usr/local/lib/python3.5/site-packages/django/contrib/admin/options.py" in wrapper
  606.                 return self.admin_site.admin_view(view)(*args, **kwargs)

File "/usr/local/lib/python3.5/site-packages/django/utils/decorators.py" in _wrapped_view
  142.                     response = view_func(request, *args, **kwargs)

File "/usr/local/lib/python3.5/site-packages/django/views/decorators/cache.py" in _wrapped_view_func
  44.         response = view_func(request, *args, **kwargs)

File "/usr/local/lib/python3.5/site-packages/django/contrib/admin/sites.py" in inner
  223.             return view(request, *args, **kwargs)

File "/app/data_import/admin.py" in change_view
  74.         processing = file.processing

File "/app/data_import/models.py" in processing
  161.                 kw_args = json.loads(task['kwargs'])

File "/usr/local/lib/python3.5/json/__init__.py" in loads
  319.         return _default_decoder.decode(s)

File "/usr/local/lib/python3.5/json/decoder.py" in decode
  339.         obj, end = self.raw_decode(s, idx=_w(s, 0).end())

File "/usr/local/lib/python3.5/json/decoder.py" in raw_decode
  355.             obj, end = self.scan_once(s, idx)

Exception Type: JSONDecodeError at /admin/data_import/standardizedfile/1/change/
Exception Value: Expecting property name enclosed in double quotes: line 1 column 2 (char 1)
```

What do you think about an eval approach instead?

Reply:
Yeah, I can try that. I'm not getting that traceback, though, and I'd like to make sure I can test the eval approach as well. When / how is this exception coming up?

Reply:
I uploaded a file, then added all responding agencies, then navigated back in my browser to the standardized file detail page.